tests/unit/source_code/test_graph.py

```diff
@@ -38,8 +38,94 @@ def test_folder_loads_content(mock_path_lookup):
         ImportFileResolver(file_loader, allow_list),
         mock_path_lookup,
     )
-    dependency = Dependency(FolderLoader(file_loader), path)
+    dependency = Dependency(FolderLoader(file_loader), path, False)
     graph = DependencyGraph(dependency, None, dependency_resolver, mock_path_lookup, session_state)
     container = dependency.load(mock_path_lookup)
     container.build_dependency_graph(graph)
     assert len(graph.all_paths) > 1
+
+
+class _TestDependencyGraph(DependencyGraph):
+    def compute_route(self, root: Path, leaf: Path) -> list[Dependency]:
+        return self._compute_route(root, leaf, set())
+
+
+def test_graph_computes_magic_run_route(mock_path_lookup, simple_dependency_resolver):
+    parent = mock_path_lookup.cwd / "functional/parent_that_magic_runs_child_that_uses_value_from_parent.py"
+    child = mock_path_lookup.cwd / "functional/_child_that_uses_value_from_parent.py"
+    dependency = Dependency(NotebookLoader(), parent)
+    root_graph = _TestDependencyGraph(
```

Comment:
you can make the constructor of _TestDependencyGraph simpler to fit test scenarios - it's already repeated 5 times, for example.

Reply:
done